Make doc/man7/ and doc/internal/man3/ conform with man-pages(7)

levitte opened the pull request:
It's all in the details, from man-pages(7):

    Formatting conventions for manual pages describing functions

        ...
        Variable names should, like argument names, be specified in italics.
        ...

    Formatting conventions (general)

        ...
        Special macros, which are usually in uppercase, are in bold.
        Exception: don't boldface NULL.
        ...

<!--
Thank you for your pull request. Please review these requirements:

Contributors guide: https://github.com/openssl/openssl/blob/master/CONTRIBUTING

Other than that, provide a description above this comment if there isn't one already

If this fixes a github issue, make sure to have a line saying 'Fixes #XXXX' (without quotes) in the commit message.
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] documentation is added or updated
- [ ] tests are added or updated

levitte committed:
Make doc/man7/ and doc/internal/man3/ conform with man-pages(7)

It's all in the details, from man-pages(7):

    Formatting conventions for manual pages describing functions

        ...
        Variable names should, like argument names, be specified in italics.
        ...

    Formatting conventions (general)

        ...
        Special macros, which are usually in uppercase, are in bold.
        Exception: don't boldface NULL.
        ...

paulidale commented on doc/man7/bio.pod:
```diff
@@ -47,7 +47,7 @@ in a memory leak.
-Normally the B<type> argument is supplied by a function which returns a
+Normally the I<type> argument is supplied by a function which returns a
 pointer to a BIO_METHOD. There is a naming convention for such functions:
 a source/sink BIO is normally called BIO_s_*() and a filter BIO
```
Way up above, `B<ossl_sa_TYPE_>` was used for a partial function name.  Why not here too?

levitte replied:
We should probably switch to what's on this line.

levitte replied:
I take that back.  Apparently, the `*` disrupts the usual function detection mechanism...  so we need to help it along.  I'll make this bold.

levitte replied:
Extra commit added.  Have a look at it, it has a proposal on how to handle function names with a variable part.

richsalz replied:
That should be backported to other "generic" manpages, or this one made to conform.
```
; git grep 'comment generic'
```

paulidale commented on doc/man7/provider-cipher.pod:
```diff
@@ -306,7 +306,7 @@ beginning of the output buffer, do "in place" encryption of the payload and
 output buffer.
 
-Whether encrypting or decrypting the value written to B<*outl> in the
+Whether encrypting or decrypting the value written to I<*outl> in the
```
Is this still a parameter name with the leading `*`?
Would `*I<outl>` be better?

richsalz replied:
Leave the pointer bolded please.

levitte replied:
Bolded?  You mean italic, I assume.  I'm torn...  is there a standard library example we can look at?

Although, I would go for `B<*>I<name>`...

paulidale replied:
`strtol` perhaps?

levitte replied:
Thanks.  Perfect example.  [strtol(3)](https://linux.die.net/man/3/strtol) shows a similar example, and the `*` are in italics (depending on your rendering, you have to examine the HTML to see it)

In other words, I leave as is.

paulidale replied:
Okay, it came out underlined here like all the other highlights.  I've no idea where MacOS stores the raw `man` pages (which aren't HTML) or I'd have checked there.

paulidale commented on doc/man7/provider-digest.pod:
```diff
@@ -99,58 +99,58 @@ OP_digest_newctx() should create and return a pointer to a provider side
-exceed B<outsz> bytes.
+passed in the I<provctx> parameter (see L<provider(3)>).
+I<inl> bytes at I<in> should be digested and the result should be stored at
+I<out>. The length of the digest should be stored in I<*outl> which should not
```
Same question about the `*`.

paulidale commented on doc/man7/provider-keyexch.pod:
```diff
@@ -87,49 +87,49 @@ OP_keyexch_newctx() should create and return a pointer to a provider side
-written to B<*secretlen>.
+The derived secret should be written to the location I<secret> which should not
+exceed I<outlen> bytes.
+The length of the shared secret should be written to I<*secretlen>.
```
And another `*` (here and two lines below).

paulidale commented on doc/man7/provider-signature.pod:
```diff
@@ -111,80 +111,80 @@ OP_signature_newctx() should create and return a pointer to a provider side
-B<*siglen>.
+Unless I<sig> is NULL, the signature should be written to the location pointed
+to by the I<sig> parameter and it should not exceed I<sigsize> bytes in length.
+The length of the signature should be written to I<*siglen>.
```
My starry fun.  I'll stop commenting on them anymore.

paulidale reviewed (commented):
Consistency is good!
A couple of mostly harmless comments.

richsalz commented:
Oh man.  This is good work, but it's sure going to be painful on all the doc-nits changes I have waiting review :(

levitte commented:
How should I say this...  I aim to please :wink:

richsalz commented:
I took a look, it won't be too bad.  Most of my pain will be self-inflicted, with #9952 and #9953 and even that's minor.

levitte commented:
See it on the bright side: you'll get to learn *real* regexps :wink:

levitte committed:
Funtion name with variable part in doc/man7/ and doc/internal/man3/

We have a few pages where part of function names can be considered
variable.  There are no normative guidelines for such a case, but if
we draw from the formatting convention of variable and argument names,
we can draw the conclusion that this variable part should be italized,
within already given conventions.  In other words, we need to help the
POD processor along in cases like these:

    SPARSE_ARRAY_OF(TYPE)
    ossl_sa_TYPE_num()

These need explicit formatting:

    B<SPARSE_ARRAY_OF>(I<TYPE>)
    B<ossl_sa_I<TYPE>_num>()

paulidale commented:
> See it on the bright side: you'll get to learn real regexps

Nah, learn something strictly more powerful.  I'm a fan of PEGs but there are others.

paulidale reviewed (commented):
Safestack and lhash could also do with the same treatment I suspect.

paulidale reviewed (approved):
I'm happy with the changes here, my prior comment can be a new PR or an extra commit here.

levitte commented:
> Safestack and lhash could also do with the same treatment I suspect.

Yes.  Do notice that it's not within the scope of this PR.  I'll make another for doc/man3/, probably piece meal as they are easier to digest.

levitte commented:
Merged.

dfabee82be84f8ac80eede5d552058fbcdf4928d Make doc/man7/ and doc/internal/man3/ conform with man-pages(7)
c18d2d94c8a27c1dcc5036bebdfcce92874a297c Funtion name with variable part in doc/man7/ and doc/internal/man3/
